library/L2_HAL/sensors/environment/temperature/test/si7060_test.cpp

```diff
@@ -1,13 +1,211 @@
+    // transaction record:
+    //   1. Write transaction to enable one-shot mode.
+    //   2. Write transaction to enable auto increment.
+    //   3. Write transaction to register pointer to most significant byte
+    //      register and retrieve the MSB of the temperature data.
+    //   4. Read transaction to retrieve the LSB of the temperature data.
+    constexpr std::array kExpectedDataOutLengths = { 2, 2, 1 };
+    constexpr uint8_t kMaxDataLength = kExpectedDataOutLengths.size();
+    // constexpr
+    const uint8_t kExpectedDataOut[][kMaxDataLength] = {
+      { Si7060::kOneBurstRegister, kEnableOneBurstMode },
+      { Si7060::kAutomaticBitRegister, kEnableAutoIncrement },
+      { Si7060::kMostSignificantRegister },
+    };
+    const std::array kExpectedTransactions = {
+      // transaction to set one-shot mode
+      I2c::Transaction_t({
+          .operation  = I2c::Operation::kWrite,
+          .address    = kDeviceAddress,
+          .data_out   = kExpectedDataOut[0],
+          .out_length = kExpectedDataOutLengths[0],
+          .data_in    = nullptr,
+          .in_length  = 0,
+          .position   = 0,
+          .repeated   = false,
+          .busy       = true,
+          .timeout    = I2c::kI2cTimeout,
+          .status     = Status::kSuccess,
+      }),
+      // transaction to enable auto increment
+      I2c::Transaction_t({
+          .operation  = I2c::Operation::kWrite,
+          .address    = kDeviceAddress,
+          .data_out   = kExpectedDataOut[1],
+          .out_length = kExpectedDataOutLengths[1],
+          .data_in    = nullptr,
+          .in_length  = 0,
+          .position   = 0,
+          .repeated   = false,
+          .busy       = true,
+          .timeout    = I2c::kI2cTimeout,
+          .status     = Status::kSuccess,
+      }),
+      // transaction to read the MSB of the temperature data
+      I2c::Transaction_t({
+          .operation  = I2c::Operation::kWrite,
+          .address    = kDeviceAddress,
+          .data_out   = kExpectedDataOut[2],
+          .out_length = kExpectedDataOutLengths[2],
+          .data_in    = nullptr,
+          .in_length  = 1,
+          .position   = 0,
+          .repeated   = true,
+          .busy       = true,
+          .timeout    = I2c::kI2cTimeout,
+          .status     = Status::kSuccess,
+      }),
+      // transaction to read the LSB of the temperature data
+      I2c::Transaction_t({
+          .operation  = I2c::Operation::kRead,
+          .address    = kDeviceAddress,
+          .data_out   = nullptr,
+          .out_length = 0,
+          .data_in    = nullptr,
+          .in_length  = 1,
+          .position   = 0,
+          .repeated   = false,
+          .busy       = true,
+          .timeout    = I2c::kI2cTimeout,
+          .status     = Status::kSuccess,
+      }),
+    };
+
+    // Array to store the two transactions.
+    static I2c::Transaction_t * transactions =
```

Comment:
```suggestion
    I2c::Transaction_t * transactions[kExpectedTransaction.size()];
```
I do not see the reason for dynamic allocation here when the length is knowable at compile time. There is also no reason to make this static either.